solr/core/src/test/org/apache/solr/cluster/placement/impl/PlacementPluginIntegrationTest.java

```diff
@@ -220,6 +231,66 @@ public void testDynamicReconfiguration() throws Exception {
+    SolrCollection collection = cluster.getCollection(COLLECTION);
+    AttributeFetcher attributeFetcher = new AttributeFetcherImpl(cloudManager);
+    String someMetricName = "solr.jvm:system.properties:user.name";
+    String sysprop = "user.name";
+    String sysenv = "PWD";
+    attributeFetcher
+        .fetchFrom(cluster.getLiveNodes())
+        .requestNodeHeapUsage()
+        .requestNodeMetric(someMetricName)
+        .requestNodeSystemProperty(sysprop)
+        .requestNodeEnvironmentVariable(sysenv)
+        .requestNodeTotalDisk()
+        .requestNodeFreeDisk()
+        .requestNodeCoresCount()
+        .requestCollectionMetrics(collection, Set.of(ReplicaMetric.QUERY_RATE_1MIN, ReplicaMetric.UPDATE_RATE_1MIN));
+    AttributeValues attributeValues = attributeFetcher.fetchAttributes();
+    String userName = System.getProperty("user.name");
+    String pwd = System.getenv("PWD");
+    // node metrics
+    for (Node node : cluster.getLiveNodes()) {
+      assertTrue("heap usage", attributeValues.getHeapUsage(node).isPresent());
+      assertTrue("total disk", attributeValues.getTotalDisk(node).isPresent());
+      assertTrue("free disk", attributeValues.getFreeDisk(node).isPresent());
+      assertTrue("cores count", attributeValues.getCoresCount(node).isPresent());
+      Optional<Object> userNameOpt = attributeValues.getNodeMetric(node, someMetricName);
+      assertTrue("user.name", userNameOpt.isPresent());
+      assertEquals("userName", userName, userNameOpt.get());
+      Optional<String> syspropOpt = attributeValues.getSystemProperty(node, sysprop);
+      assertTrue("sysprop", syspropOpt.isPresent());
+      assertEquals("user.name sysprop", userName, syspropOpt.get());
+      Optional<String> sysenvOpt = attributeValues.getEnvironmentVariable(node, sysenv);
+      assertTrue("sysenv", sysenvOpt.isPresent());
+      assertEquals("PWD sysenv", pwd, sysenvOpt.get());
+    }
+    assertTrue(attributeValues.getCollectionMetrics(COLLECTION).isPresent());
+    CollectionMetrics collectionMetrics = attributeValues.getCollectionMetrics(COLLECTION).get();
+    collection.shards().forEach(shard -> {
+      Optional<ShardMetrics> shardMetricsOpt = collectionMetrics.getShardMetrics(shard.getShardName());
+      assertTrue("shard metrics", shardMetricsOpt.isPresent());
+      shard.replicas().forEach(replica -> {
+        Optional<ReplicaMetrics> replicaMetricsOpt = shardMetricsOpt.get().getReplicaMetrics(replica.getReplicaName());
+        assertTrue("replica metrics", replicaMetricsOpt.isPresent());
+        ReplicaMetrics replicaMetrics = replicaMetricsOpt.get();
+        // this should always be present
+        assertNotNull("size", replicaMetrics.getReplicaSizeGB());
+        assertTrue("should be greater than 0 but was " + replicaMetrics.getReplicaSizeGB(),
+            replicaMetrics.getReplicaSizeGB() > 0);
+        assertNotNull("queryRate", replicaMetrics.getReplicaMetric(ReplicaMetric.QUERY_RATE_1MIN.getName()));
```

Comment:
We need to retrieve the value of a replica metric and verify it (previously it would have been making sure it is cast to the appropriate type, but with the ongoing changes, show how these values are read and make sure things actually works).

Reply:
This will possibly happen earlier in the test code if index size is converted into a normal metric though.